src/components/data-display/tables/ReturnedNamesTable.tsx

```diff
@@ -0,0 +1,484 @@
+        const { name, initiator, endTimestamp } = nameData;
+        const data: TableData = {
+          openRow: <></>,
+          name,
+          closingDate: endTimestamp,
+          initiator,
+          leasePrice: -1,
+          permabuy: -1,
+          returnType:
+            initiator === arioProcessId ? 'Lease expiry' : 'Permanent Return',
+
+          action: <></>,
+          // metadata used for search and other purposes
+          returnNameData: nameData,
+        };
+        newTableData.push(data);
+      });
+
+      setTableData(newTableData);
+    }
+  }, [returnedNames, loading]);
+  async function fetchPrice(
+    name: string,
+    type: TRANSACTION_TYPES,
+  ): Promise<number | Error> {
+    try {
+      const res = await queryClient.fetchQuery(
+        buildCostDetailsQuery(
+          {
+            intent: 'Buy-Record',
+            name,
+            type,
+            years: 1,
+            fromAddress: walletAddress?.toString(),
+          },
+          { arioContract, arioProcessId },
+        ),
+      );
+
+      return res.tokenCost;
+    } catch (error: any) {
+      return new Error(error.message);
+    }
+  }
+  useEffect(() => {
+    async function updatePrices() {
+      // Filter rows that need price updates
+      const rowsToUpdate = tableData.filter(
+        (row) =>
+          row.leasePrice instanceof Error ||
+          row.leasePrice < 0 ||
+          row.permabuy instanceof Error ||
+          row.permabuy < 0,
+      );
+
+      if (rowsToUpdate.length === 0) {
+        // No rows need updates, exit early
+        return;
+      }
+
+      const updatedData = await Promise.all(
+        tableData.map(async (row) => {
+          if (
+            row.leasePrice instanceof Error ||
+            row.leasePrice < 0 ||
+            row.permabuy instanceof Error ||
+            row.permabuy < 0
+          ) {
+            // Fetch lease price
+            const leasePrice = await fetchPrice(
+              row.name,
+              TRANSACTION_TYPES.LEASE,
+            );
+            // Fetch permabuy price
+            const permabuyPrice = await fetchPrice(
+              row.name,
+              TRANSACTION_TYPES.BUY,
+            );
+
+            // Return updated row with fetched prices
+            return {
+              ...row,
+              leasePrice,
+              permabuy: permabuyPrice,
+            };
+          }
+          // Return the row unchanged if no update is needed
+          return row;
+        }),
+      );
+
+      // Set updated table data
+      setTableData(updatedData);
+    }
+
+    updatePrices();
+  }, [tableData]); // Re-run only when `tableData` changes
+
+  useEffect(() => {
+    if (filter) {
+      setFilteredTableData(filterTableData(filter, tableData));
+    } else {
+      setFilteredTableData([]);
+    }
+  }, [filter, tableData]);
+  // Define columns for the table
+  const columns: ColumnDef<TableData, any>[] = [
```

Comment:
I find this mapping here to be less clear vs. just using code for each column def entry since you have lots of conditionals below. Not much saved and harder to read.

Reply:
I like it, its nice to have all the headers at the top for making new tables. I can just copy paste the config and then edit them all in one place.

Reply:
Not a fan of the map. Can copy/paste configs without using map.

Reply:
Well there 3-4 other tables that have this setup, so ill make a ticket for a refactor as tech debt